### PR TITLE
ImageEdit: Support ModeIndependentColor for the border color

### PR DESCRIPTION
**Pull request by haven2world**
## Changes
Change the `borderColor` of `ImageEditOptions` from `string` to `ModeIndependentColor | string`

### CODE REVIEW

File: packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts [hunk 1]
```diff
@@ -70,7 +70,10 @@ const FeatureToOperationMap = {
  * Default image edit options
  */
 const DefaultOptions: Required<ImageEditOptions> = {
-    borderColor: '#DB626C',
+    borderColor: {
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts:73]: And when use string|ModeIndenpendColor, no need to change here

File: packages/roosterjs-editor-types/lib/interface/ImageEditOptions.ts [hunk 2]
```diff
@@ -6,7 +8,7 @@ export default interface ImageEditOptions {
      * Color of resize/rotate border, handle and icon
      * @default #DB626C
      */
-    borderColor?: string;
+    borderColor?: ModeIndependentColor;
```
[the remark this answers is not in the document]
Comment (haven2world) [reply packages/roosterjs-editor-types/lib/interface/ImageEditOptions.ts]: Good point!

File: packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts [hunk 3]
```diff
@@ -382,7 +385,7 @@ export default class ImageEdit implements EditorPlugin {
 
         // Get HTML for all edit elements (resize handle, rotate handle, crop handle and overlay, ...) and create HTML element
         const options: ImageHtmlOptions = {
-            borderColor: this.options.borderColor,
+            borderColor: this.editor.isDarkMode ? this.options.borderColor.darkModeColor : this.options.borderColor.lightModeColor,
```
[the remark this answers is not in the document]
Comment (haven2world) [reply packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts]: Good catch! Fixed it

**Comment by haven2world** [comment]
If it looks good, please help merge it. I don't have permission. Thanks!

File: packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts [hunk 4]
```diff
@@ -671,3 +672,10 @@ function isASmallImage(editInfo: ImageEditInfo, isFeatureEnabled?: boolean) {
     const { widthPx, heightPx } = editInfo;
     return widthPx && heightPx && widthPx * widthPx < MAX_SMALL_SIZE_IMAGE && isFeatureEnabled;
 }
+
+function getColorString(color: string | ModeIndependentColor, isDarkMode: boolean): string {
+    if (typeof color === 'string') {
+        return color.trim();
+    }
+    return isDarkMode ? color.darkModeColor.trim() : color.lightModeColor.trim();
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts:680]: What if the color is undefined?

Comment (JiuqingSong) [reply packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts:680]: @haven2world would you please take a look at my comment? thanks.

Comment (haven2world) [reply packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts:680]: I replied but I don't know why you cannot see it
image

Comment (haven2world) [reply packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts:680]: Sorry, looks like I didn't commit the comment... I'm not familiar with the workflow on GitHub. Haha it's kind of different from ADO

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**